### ISSUE TITLE
Mixed nodes in XML are out of order

### ISSUE THREAD

**Issue by detomastah** [YorickPeterse/oga#74]
In the mixed node (element with text and elements inside) all the non-text elements are at the beginning of the parent.

Original doc
```
<parent>
  dog <chow></chow> cow
</parent>
```
```
Result from xpath query / document.to_xml

<parent>
  <chow></chow> dog  cow
</parent>
```

**Comment by YorickPeterse** [comment]
I'm not really following, the following works fine:

```ruby
doc = Oga.parse_xml <<-EOF
<parent>
  dog <chow></chow> cow
</parent>
EOF

puts doc.to_xml
```

This outputs:

```xml
<parent>
  dog <chow /> cow
</parent>
```

Do you have a way to reproduce whatever problem you're trying to describe?

**Comment by detomastah** [comment]
Thanks for really quick response. I would like to fix the problem on my own, even peeked at the code and grammars...

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-stylesheet type="text/xsl" href="../../../xml/stylesheets/page.xsl"?>
<document uid="RUBY-000" title="Nasze relacje z komputerami" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../../../xml/schemas/page.xsd">

  <section title="Jak myśleć o programach?">
    <p>
      W przypadku powyższego programu prawdopodobieństwo wyświetlenia wzrasta! Słowo <exp>print</exp> po angielsku oznacza "drukuj". Jednak nadal nie wiesz czy na pewno tak będzie. Być może słowo <exp>print</exp> w tym programie oznacza włączenie żarówki o numerze 10. Nie dowiesz się tego, dopóki nie poznasz specyfikacji języka.
    </p>
  </section>
</document>
```

This is the xml which does not render correctly. On Nokogiri I have the other problem, after some processing when I call .to_s all the `<exp></exp>` tags are removed. That was my motive to use Oga. In oga, in the same example it puts all the exp's at the beginning of the content, so I got:

```xml
<exp>print</exp><exp>print</exp>The rest of the text
```

**Comment by YorickPeterse** [comment]
@detomastah I'm sorry but I don't understand what is going on. If you could provide a small _Ruby script_ that reproduces whatever problem you are experiencing that would greatly help.

**Comment by detomastah** [comment]
Sorry for taking your time. I found the problem - it was my fault. One of the routines was doing something like this:

```ruby
doc.xpath("//p").each do |el|
  el.inner_text = el.inner_text.strip
end
```

**Comment by YorickPeterse** [comment]
@detomastah Ah, that particular problem has been solved in #64.

**Comment by detomastah** [comment]
Thanks!
